dbms/src/Operators/LimitTransform.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Operators/LimitTransform.h>
+
+namespace DB
+{
+void GlobalLimitTransformAction::transform(Block & block)
+{
+    if (unlikely(!block))
+        return;
+
+    /// pos - how many lines were read, including the last read block
+    if (pos >= limit)
+    {
+        block = {};
```

Comment:
refine `LimitTransformAction` line 46 is enough.

Reply:
ok, done